### PR TITLE
feat(aws-lambda-*): add custom environment variable name

### PR DESCRIPTION
**Pull request by beomseoklee**
*Issue #, if available:*
#132 

*Description of changes:*
I've added custom environment variable name to every `aws-lambda-*` pattern based on the recommendation on #132 .

- added custom environment variable name to all `aws-lambda` patterns
- updated `README`
- enhanced unit test cases

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-QfXHf1ULVFb1
* Commit ID: d71d31387c94b52b78223e3f62fd6c88ef8f4713
* Result: FAILED
* [Build Logs](https://idenu9lil3.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c05efbac-8864-4e08-b32b-84468f3e0c24%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-QfXHf1ULVFb1
* Commit ID: eee097985e29a0b34c1056527fe82a783d256ad4
* Result: SUCCEEDED
* [Build Logs](https://idenu9lil3.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=68f6f2c8-cab4-4a55-9df4-18f98661acd6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — CHANGES_REQUESTED** [review]

File: source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/README.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ _Parameters_
 |dynamoTableProps?|[`dynamodb.TableProps`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-dynamodb.TableProps.html)|Optional user provided props to override the default props for DynamoDB Table|
 |existingTableObj?|[`dynamodb.Table`](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-dynamodb.Table.html)|Existing instance of DynamoDB table object, If this is set then the dynamoTableProps is ignored|
 |tablePermissions?|`string`|Optional table permissions to grant to the Lambda function. One of the following may be specified: `All`, `Read`, `ReadWrite`, `Write`.|
+|tableEnvironmentVariableName?|`string`|Optional Lambda function environment variable for the DynamoDB table.|
```
[the remark this answers is not in the document]
Comment (beomseoklee) [reply source/patterns/@aws-solutions-constructs/aws-lambda-dynamodb/README.md]: Updated.

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-QfXHf1ULVFb1
* Commit ID: 284be19c01b6dd73d1d8e4c5bd74b16b1dfa4e50
* Result: SUCCEEDED
* [Build Logs](https://idenu9lil3.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=cb05ee7e-3ee6-42af-9165-3b688ec5423e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 284be19c01b6dd73d1d8e4c5bd74b16b1dfa4e50
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c9fe116f-e49a-41c1-9e90-f7ae8bd7ab11%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-QfXHf1ULVFb1
* Commit ID: ccbb406cf3b7b56ce32eeafba99c73e32b68553a
* Result: SUCCEEDED
* [Build Logs](https://idenu9lil3.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=364d591b-1d3f-403e-8ee1-0f482edf1eab%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: ccbb406cf3b7b56ce32eeafba99c73e32b68553a
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=904e1da4-e934-4861-bb8a-7368ec33f047%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**